src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -874,6 +874,10 @@ void prepare_buffer_fusing::run(program& p) {
                     }
                 }
             }
+            // For better performance, disable buffer fusing with data padding which causes selecting gemm_ref kernel instead of gemm_tiled_opt
+            if (node.get_users().front()->is_type<gemm>() && crop_layout.data_padding)
```

Comment:
I dont think this makes sense, which is canceling buffer fusing, after doing all buffer fusing action for any gemm. 
I think we need to investigate what exact condition of the specific case which causes the target problem.

Reply:
Updated